.github/workflows/pull-request.yml

```
@@ -76,101 +87,117 @@ jobs:
-      - name: Changed Files
-        id: changed-files
-        uses: tj-actions/changed-files@v45
-      - name: List all changed files
-        run: |
-          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
-            echo "$file was changed"
-          done
-      - name: Check if the source and target branches are different
+      - name: Check if branches are different
         if: ${{ github.event.pull_request.base.ref == github.event.pull_request.head.ref }}
         run: |
```

actionlint (1.7.4)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)